### PR TITLE
Print content of columns for gtest

### LINKED ISSUE #5203: Print content of columns for gtest
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

Print content of columns for gtest. BTW, It's just for the convenient of debug, so it can only be used in gtest.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #5203 

Problem Summary:

### What is changed and how it works?

Results can be printed as the following:
```
printColumns({expected_col, actual_col});

col4: (0: NULL, 1: Int64_-234, 2: Int64_-123, 3: Int64_0, 4: Int64_5, 5: Int64_9999, 6: Int64_24353)
__QB_1___QB_2_col4: (0: NULL, 1: Int64_-234, 2: Int64_-123, 3: Int64_0, 4: Int64_5, 5: Int64_9999, 6: Int64_24353)
```
```
printColumns({expected_col, actual_col}, 1, 4);

col4: (1: Int64_-234, 2: Int64_-123, 3: Int64_0, 4: Int64_5)
__QB_1___QB_2_col4: (1: Int64_-234, 2: Int64_-123, 3: Int64_0, 4: Int64_5)
```
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/TestUtils/FunctionTestUtils.cpp [hunk 1]
```diff
@@ -275,5 +275,54 @@ ColumnWithTypeAndName toNullableDatetimeVec(String name, const std::vector<Strin
...
+                output = fmt::format("{}, ", output); /// Add comma to seperate different values
+
+            /// Add value
+            output = fmt::format("{}{}: {}", output, j, (*cols[i].column)[j].toString());
+        }
+
+        output = fmt::format("{})\n", output);
+    }
+
+    std::cout << output << std::endl;
```
Comment (JaySon-Huang) [inline dbms/src/TestUtils/FunctionTestUtils.cpp:324]: Use Logger? Or just make this function return a string and let the caller decide how to print it?

Comment (xzhangxian1008) [reply dbms/src/TestUtils/FunctionTestUtils.cpp]: > Use Logger? Or just make this function return a string and let the caller decide how to print it?

I have modified it to return a string.

File: dbms/src/TestUtils/FunctionTestUtils.cpp [hunk 2]
```diff
@@ -375,9 +377,63 @@ ColumnWithTypeAndName toNullableDatetimeVec(String name, const std::vector<Strin
...
 }
 
+String getColumnsContent(const ColumnsWithTypeAndName & cols)
+{
+    if (cols.size() <= 0)
+        return "";
+    return getColumnsContent(cols, 0, cols[0].column->size() - 1);
+}
+
+String getColumnsContent(const ColumnsWithTypeAndName & cols, size_t begin, size_t end)
```
Comment (ywqzzy) [inline dbms/src/TestUtils/FunctionTestUtils.cpp:387]: maybe add a unit test for the function？

Comment (xzhangxian1008) [reply dbms/src/TestUtils/FunctionTestUtils.cpp:387]: > maybe add a unit test for the function？

Done

**Review by ywqzzy — APPROVED** [review]
LGTM

File: dbms/src/TestUtils/FunctionTestUtils.cpp [hunk 3]
```diff
@@ -375,9 +377,63 @@ ColumnWithTypeAndName toNullableDatetimeVec(String name, const std::vector<Strin
...
+            col_content.push_back(std::make_pair(j, (*cols[i].column)[j].toString()));
+
+        /// Add content
+        fmt_buf.joinStr(
+            col_content.begin(),
+            col_content.end(),
+            [](const auto & content, FmtBuffer & fmt_buf) {
+                fmt_buf.append(fmt::format("{}: {}", content.first, content.second));
+            },
+            ",");
```
Comment (ywqzzy) [inline dbms/src/TestUtils/FunctionTestUtils.cpp:424]:
```suggestion
            ", ");
```
Maybe better？

**Review by SeaRise — APPROVED** [review]
/merge

**Pull request merged**